### PR TITLE
Fix crash on authentification error for a zone.

### PR DESCRIPTION
**Pull request by cpaillet**
If an api token don't have access to a zone, analytics.colos raise a
CloudFlareAPIError.i
we just don't expose this zone now.

### CODE REVIEW

File: cloudflare_exporter/collector.py [hunk 1]
```diff
@@ -86,7 +86,22 @@ def collect(self):
 
 def _get_cloudflare_analytics(token):
     cloudflare = CloudFlare.CloudFlare(debug=False, token=token)
-    for zone in cloudflare.zones.get():
+    try:
+        zones = cloudflare.zones.get()
+    except CloudFlare.exceptions.CloudFlareAPIError as e:
+        exit('/zones.get %d %s - api call failed' % (e, e))
+    except Exception as e:
+        exit('/zones.get - %s - api call failed' % (e))
```
Comment (pierrecdn) [inline cloudflare_exporter/collector.py]: This is a bad pattern (too often used), it should fail the whole process.

File: cloudflare_exporter/collector.py [hunk 2]
```diff
@@ -86,7 +86,22 @@ def collect(self):
...
+        try:
+            series = cloudflare.zones.analytics.colos(zone['id'], params={'since': -35, 'until': -5})
+        except CloudFlare.exceptions.CloudFlareAPIError as e:
+            code, string = e.args
+            if code == 10000:
+                # Authentication error on this zone only
+                continue
+            exit('/zones.get %d %s - api call failed' % (e, e))
+        except Exception as e:
+            exit('/zones.get - %s - api call failed' % (e))
```
Comment (pierrecdn) [inline cloudflare_exporter/collector.py]: Same here

**Review by pierrecdn — COMMENTED** [review]
I don't fully understand you commit body:
* "CloudFlareAPIError.i": typo?
* "we just don't expose this zone now.": ?

File: cloudflare_exporter/collector.py (hunk 1 above)
Comment (cpaillet) [reply cloudflare_exporter/collector.py]: I rewrite this part

Comment (cpaillet) [reply cloudflare_exporter/collector.py]: If a user ( token) don't have acces to a cloudflare zone ( like "criteo.com" ), we don't want to crash, but export others zones

**Comment by cpaillet** [comment]
I will create a new one

**Pull request closed without merging**